### PR TITLE
[PM-8985] Distribute beta builds on every push to main

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8985

## 📔 Objective

Every push to main will draft a new Beta release to the Closed Beta Testing track in Google Play Store.

Unless a version name is provided when manually triggering this job, the version name will be calculated based on the latest git tag.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -368,8 +368,8 @@ jobs:
           bundle exec fastlane run validate_play_store_json_key
 
       - name: Publish Play Store bundle
-        if: ${{ matrix.variant == 'prod' && matrix.artifact == 'aab' && inputs.publish-to-play-store }}
-        run: bundle exec fastlane publishForInternalTesting
```
Comment (david-livefront) [inline .github/workflows/build.yml:372]: Do we need some other task to publish it for internal testing still?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -126,7 +126,28 @@ platform :android do
     currentVersionName = buildConfigText.match(/versionName = "(.+)"/).captures[0]
 
     if options[:versionName].nil? or options[:versionName].to_s.empty?
+        puts "Getting latest version name from previous git tag..."
+        latestTag = `git tag --sort=committerdate --list | tail -1`
+        puts "Using last tag (#{latestTag}) to calculate version name..."
+        latestTag.slice!(0)
+        puts "Current version name resolved to #{latestTag}"
+
         nextVersionName = currentVersionName
```
Comment (david-livefront) [inline fastlane/Fastfile:135]: Can we remove this line?

File: .github/workflows/build.yml (hunk 1 above)
Comment (SaintPatrck) [reply .github/workflows/build.yml:372]: No. At the moment we're only publishing to the Public Beta. Internal testing will be done using the internal Firebase distributions.

Comment (david-livefront) [reply .github/workflows/build.yml:372]: OK

**Review by david-livefront — APPROVED** [review]

**Pull request merged**